libcudacxx/include/cuda/std/detail/libcxx/include/__cuda/barrier.h

```diff
@@ -857,4 +857,180 @@ async_contract_fulfillment memcpy_async(void * __destination, void const * __sou
 
 _LIBCUDACXX_END_NAMESPACE_CUDA
 
+_LIBCUDACXX_BEGIN_NAMESPACE_CUDA_EXPERIMENTAL
```

Comment:
Since these are device-only functions, the namespace should be `cuda::device::experimental::`.